### ISSUE TITLE
Possible CPU usage spikes

### ISSUE THREAD

**Issue by timkendall** [apollographql/apollo-server#1528]
Just upgraded our large GraphQL service to `apollo-server-koa@2` and am seeing inconsistent CPU spikes all over the place. This seems to have caused inconsistent query response times (sometime queries are fast and sometimes they take many seconds).

Haven't had a chance to dig in yet but figured it was worth reporting now.  Worth noting that we rely heavily on remote schema stitching. 

Mega Spikes:

![image](https://user-images.githubusercontent.com/1427378/44063006-8732aa18-9f13-11e8-8a89-e66e48c0a3ea.png)

**Comment by timkendall** [comment]
Graphql.js 14.0.0 may fix this it seems, see:

- https://github.com/graphql/graphql-js/pull/1417
- https://github.com/graphql/graphql-js/pull/1427

**Comment by timkendall** [comment]
Ugh thats gnarly. @human-a what version of node are you guys on (we're on `v10.9.0`)?

**Comment by human-a** [comment]
@timkendall Yes, it surely is gnarly. We are on ```v8.12.0```
I have also contacted AWS support (we have the business support plan) and they are looking into the issue to find the source of the issue. I will reply back here if they give me something useful, you may find it useful too.

**Comment by ZhimaoL** [comment]
I am wondering have u guys solved the CPU problems? I tried to run apollo server as API gateway (simple logic in apollo server) on kubernetes, and experienced a very high CPU usage. Especially the socket connections with redis pubsub, the CPU will reach 1.5 core quickly. Do you guys have any related experiences? @human-a @timbotnik

**Comment by human-a** [comment]
@ZhimaoL In our case, the issue was not related to GraphQL, it was caused by another library we were using, which was causing infinite loops, and because NodeJS is rather dumb, we could not see any errors or warnings. All I did was to use the library on the client side, and bam, there was the error on our console.

The library that was causing the issue: https://github.com/TinyNova/deep-omit-by
We fixed the infinite loop it by doing so ```JSON.parse(JSON.stringify(ourObject))```

I hope this helps. I would suggest looking elsewhere for the source of your CPU spikes first.
